File: client/src/types/ai-stream.ts
export type AIStreamEventType =
  | "started"
  | "progress"
  | "retrieval_started"
  | "retrieval_completed"
  | "question_generated"
  | "question_validated"
  | "difficulty_scored"
  | "bloom_scored"
  | "analytics_generated"
  | "recommendations_generated"
  | "completed"
  | "error";

export interface AIStreamEvent<T = unknown> {
  eventId: string;
  sessionId: string;
  timestamp: string;
  type: AIStreamEventType;
  node: string;
  progress: number;
  payload?: T;
  error?: string;
}

File: server/internal/dto/ai_stream.go
package dto

import "time"

type AIStreamEventType string

const (
	EventStarted                 AIStreamEventType = "started"
	EventProgress                AIStreamEventType = "progress"
	EventRetrievalStarted        AIStreamEventType = "retrieval_started"
	EventRetrievalCompleted      AIStreamEventType = "retrieval_completed"
	EventQuestionGenerated       AIStreamEventType = "question_generated"
	EventQuestionValidated       AIStreamEventType = "question_validated"
	EventDifficultyScored        AIStreamEventType = "difficulty_scored"
	EventBloomScored             AIStreamEventType = "bloom_scored"
	EventAnalyticsGenerated      AIStreamEventType = "analytics_generated"
	EventRecommendationsGenerated AIStreamEventType = "recommendations_generated"
	EventCompleted               AIStreamEventType = "completed"
	EventError                   AIStreamEventType = "error"
)

type AIStreamEvent struct {
	EventID   string            `json:"eventId"`
	SessionID string            `json:"sessionId"`
	Timestamp time.Time         `json:"timestamp"`
	Type      AIStreamEventType `json:"type"`
	Node      string            `json:"node"`
	Progress  float64           `json:"progress"`
	Payload   interface{}       `json:"payload,omitempty"`
	Error     string            `json:"error,omitempty"`
}
